server/src/main/java/org/apache/iotdb/db/query/executor/QueryRouter.java

```diff
@@ -225,12 +225,8 @@ public QueryDataSet groupByFill(GroupByTimeFillPlan groupByFillPlan, QueryContex
   @Override
   public QueryDataSet lastQuery(LastQueryPlan lastQueryPlan, QueryContext context)
       throws StorageEngineException, QueryProcessException, IOException {
-    LastQueryExecutor lastQueryExecutor = getLastQueryExecutor(lastQueryPlan);
+    LastQueryExecutor lastQueryExecutor = new LastQueryExecutor(lastQueryPlan);
     return lastQueryExecutor.execute(context, lastQueryPlan);
   }
 
-  protected LastQueryExecutor getLastQueryExecutor(LastQueryPlan lastQueryPlan) {
```

Comment:
this may be used in cluster

Reply:
Yes, indeed. I hope you can make some interfaces out of it, which would better not be modified so frequently.